test/browser-tests/precache/precache.js

```diff
@@ -0,0 +1,175 @@
+/*
+  Copyright 2014 Google Inc. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+// This is a test and we want descriptions to be useful, if this
+// breaks the max-length, it's ok.
+
+/* eslint-disable max-len, no-lonely-if */
+/* eslint-env browser, mocha */
+/* global testHelper */
+
+'use strict';
+
+describe('Test precache method', () => {
+  let compareCachedAssets = function(assetList, cachedAssets) {
+    return new Promise((resolve, reject) => {
+      var cachedAssetsKeys = Object.keys(cachedAssets);
+      cachedAssetsKeys.should.have.length(assetList.length);
```

Comment:
If there are repeated entries in `assetList`, then it's possible that `assetList` and `cachedAssetKeys` will have the same length despite there being more unique entries in `cachedAssetKeys`.

If you're don't care about testing scenarios in which `cachedAssetKeys` has additional entries that aren't accounted for in `assetList`, then you should remove this assertion.

If you do care about testing that `assetList` and `cachedAssetKeys` are exactly the same, then you should remove duplicates from `assetList` before asserting on the lengths. Using a `Set` might help.